src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationAnalysisValueProvider.cs

```diff
@@ -42,7 +42,18 @@ internal bool TryGetValue(TKey key, out TValue value)
             // Store the value for the lifetime of the compilation.
             lock (_valueMap)
             {
-                _valueMap[key] = value;
+                // Check if another thread already stored the computed value.
```

Comment:
@sharwell This should address your comment: https://github.com/dotnet/roslyn/pull/8375/files#r52127390